From 08a02077c7a23060f233956b682319220666abd9 Mon Sep 17 00:00:00 2001
From: Chienyuan Huang <chienyuanhuang@google.com>
Date: Wed, 9 Feb 2022 06:38:48 +0000
Subject: [PATCH] Fix OOB of L2CAP_CMD_CREDIT_BASED_CONN_RES

Change-Id: Ic947572e179ec4dacae9655192cadac7cf8c3ea4
TAG: #security
Bug: 213519176
Test: manual
Ignore-AOSP-First: security fix
(cherry picked from commit ed4b512d2b3d45151b7f70e4c8c60b1a052a5d7c)
Merged-In: Ic947572e179ec4dacae9655192cadac7cf8c3ea4
---
 stack/l2cap/l2c_ble.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/stack/l2cap/l2c_ble.cc b/stack/l2cap/l2c_ble.cc
index 16454a5b0..d5486794d 100644
--- a/stack/l2cap/l2c_ble.cc
+++ b/stack/l2cap/l2c_ble.cc
@@ -621,7 +621,7 @@ void l2cble_process_sig_cmd(tL2C_LCB* p_lcb, uint8_t* p, uint16_t pkt_len) {
       break;
     }
     case L2CAP_CMD_CREDIT_BASED_CONN_RES:
-      if (p + 2 > p_pkt_end) {
+      if (p + 8 > p_pkt_end) {
         LOG(ERROR) << "invalid L2CAP_CMD_CREDIT_BASED_CONN_RES len";
         return;
       }
-- 
2.35.1.894.gb6a874cedc-goog

